packages/serverpod_serialization/lib/src/serialization.dart

```diff
@@ -209,6 +213,9 @@ abstract class SerializationManager {
               .map((e) => {'k': e.key, 'v': e.value})
               .toList();
         } else {
+          if (checkIfIsProtocol && nonEncodable is ProtocolSerialization) {
```

Comment:
Suggestion: Remove nested if statement.

Better to have this branch on the same level as the rest and then have the else statement at the bottom with the regular `toJson` method.